lightningd/channel_control.c

```diff
@@ -2462,3 +2462,63 @@ static const struct json_command dev_quiesce_command = {
 	.dev_only = true,
 };
 AUTODATA(json_command, &dev_quiesce_command);
+
+static struct command_result *json_alt_addr(struct command *cmd,
+					    const char *buffer,
+					    const jsmntok_t *obj UNNEEDED,
+					    const jsmntok_t *params)
+{
+	struct node_id *p_id;
+	struct peer *peer;
+	struct channel *channel;
+	const jsmntok_t *alt_addrs_tok;
+	bool more_than_one;
+
+	if (!param_check(cmd, buffer, params,
+			 p_req("node_id", param_node_id, &p_id),
+			 p_req("alt_addrs", param_array, &alt_addrs_tok),
+			 NULL))
+		return command_param_failed();
+
+	peer = peer_by_id(cmd->ld, p_id);
+	if (!peer) {
+		return command_fail(cmd, JSONRPC2_INVALID_REQUEST,
+				    "No such peer: %s",
+				    fmt_node_id(cmd, p_id));
+	}
+
+	channel = peer_any_channel(peer, channel_state_can_add_htlc, &more_than_one);
+	if (!channel || !channel->owner)
+		return command_fail(cmd, LIGHTNINGD, "Peer bad state");
+	/* This is a dev command: fix the api if you need this! */
+	if (more_than_one)
+		return command_fail(cmd, LIGHTNINGD, "More than one channel");
+
```

Comment:
No it's not a dev command!

Plus, you can totally whitelist ids which are not peers.